app/lib/submission_builder/ty2022/states/ny/individual_return.rb

```diff
@@ -44,38 +44,38 @@ def documents_wrapper
               end
 
               xml.tiPrime do
-                xml.FIRST_NAME @submission.data_source.primary.first_name if @submission.data_source.primary.first_name.present?
-                xml.MI_NAME @submission.data_source.primary.middle_initial if @submission.data_source.primary.middle_initial.present?
-                xml.LAST_NAME @submission.data_source.primary.last_name if @submission.data_source.primary.last_name.present?
+                xml.FIRST_NAME @submission.data_source.primary.first_name&.strip.gsub(/\s+/, ' ') if @submission.data_source.primary.first_name.present?
+                xml.MI_NAME @submission.data_source.primary.middle_initial&.strip.gsub(/\s+/, ' ') if @submission.data_source.primary.middle_initial.present?
+                xml.LAST_NAME @submission.data_source.primary.last_name&.strip.gsub(/\s+/, ' ') if @submission.data_source.primary.last_name.present?
                 if @submission.data_source.direct_file_data.mailing_street.present?
                   process_mailing_street(xml)
                 end
                 xml.MAIL_CITY_ADR @submission.data_source.direct_file_data.mailing_city.slice(0, 18) if @submission.data_source.direct_file_data.mailing_city.present?
-                xml.MAIL_STATE_ADR @submission.data_source.direct_file_data.mailing_state if @submission.data_source.direct_file_data.mailing_state.present?
+                xml.MAIL_STATE_ADR @submission.data_source.direct_file_data.mailing_state.strip.gsub(/\s+/, ' ') if @submission.data_source.direct_file_data.mailing_state.present?
                 xml.MAIL_ZIP_5_ADR @submission.data_source.direct_file_data.mailing_zip.slice(0, 5) if @submission.data_source.direct_file_data.mailing_zip.present?
-                xml.COUNTY_CD @submission.data_source.county_code if @submission.data_source.county_code.present?
-                xml.COUNTY_NAME @submission.data_source.county_name&.truncate(20) if @submission.data_source.county_name.present?
+                xml.COUNTY_CD @submission.data_source.county_code.strip.gsub(/\s+/, ' ') if @submission.data_source.county_code.present?
+                xml.COUNTY_NAME @submission.data_source.county_name.slice(0, 20) if @submission.data_source.county_name.present?
```

Comment:
actually I think the order should be gsub, slice, strip in case there's a double white space early in the string that can be shortened before the slicing